splunk_add_on_ucc_framework/generators/conf_files/conf_generator.py

```diff
@@ -0,0 +1,47 @@
+from abc import abstractmethod
+from typing import Any, Dict
+
+from splunk_add_on_ucc_framework.global_config import GlobalConfig
+
+from ..file_generator import FileGenerator
+
+
+class ConfGenerator(FileGenerator):
```

Comment:
Do we actually need this class between the base class and the actual subclasses? It only introduces `generate_conf` and `generate_conf_spec` methods and calls them. But I see not every conf file needs the spec.

Reply:
Yes, from code, this class is redundant. However, my thought process in the design were to create a hierarchy of classes with justifiable bifurcation so that it is easier to maintain the procedures of file that are being generated and the order they should be generated in.